### PR TITLE
LICENSING: Adding headers

### PR DESCRIPTION
**Pull request by splovyt**
Completed the LICENSE file since most models have left this blank.

### CODE REVIEW

**Comment by djalova** [comment]
I think we're actually supposed to leave that blank. @xuhdev

**Comment by xuhdev** [comment]
Yes. Actually https://github.com/kubernetes/kubernetes/commit/d30db1f9a915aa95402e1190461469a1889d92be is pretty illustrative and offers a pretty clear explanation and reference.

**Comment by splovyt** [comment]
Interesting. That means that we should have headers for all of our files instead. Uh oh!

**Comment by splovyt** [comment]
I'll leave it open for a little bit longer to have other people come across this for information, but will close the PR soon after.

**Comment by xuhdev** [comment]
I agree with @ajbozarth . It's the best practice to claim copyright of all source files with the least ambiguity. For example, if someone picked a single source file out from one of our repos without visiting the repo (such as from a link posted somewhere), this person may not be aware of the owner and license. Furthermore, if this person later infringes the copyright for that file, they would have one more factor against us (not clearly stated copyright notice and license).

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2019, IBM (Center of Open-Source Data & AI Technologies)
```
Comment (ajbozarth) [inline LICENSE]: Separate from the current discussion, all IBM licenses should use the same copyright (ie not our team), so this (and any header) should instead be:
```
Copyright 2018 IBM Corp. All Rights Reserved.
```

**Comment by xuhdev** [comment]
I should also note that software projects of Apache foundation themselves keep the license intact. For example, see [httpd's license](http://svn.apache.org/viewvc/httpd/httpd/trunk/LICENSE?revision=1824973&view=markup) (scroll to the end of the Apache license part)

File: LICENSE (hunk 1 above)
Comment (I-Hong) [reply LICENSE]: Minior question, is it Center `of` Open-Source or Center `for` Open-Source? 
https://developer.ibm.com/code/open/centers/codait/about/

**Comment by splovyt** [comment]
I updated the LICENSE file with Alex' comment, but we won't merge this right?

File: LICENSE (hunk 1 above)
Comment (ajbozarth) [reply LICENSE]: Nice catch I-Hong, though this has been removed, it's good to remember that it's Center *FOR* Open-Source

**Comment by splovyt** [comment]
I'll do it!

**Comment by xuhdev** [comment]
I think @ajbozarth has missed the story. The main argument here is that the whole change to `LICENSE` should be abandoned, regardless what copyright header to be used. There are plenty of commits reverting exactly what this PR does in the open source community, as I showed in a previous comment, the example in [kubernetes](https://github.com/kubernetes/kubernetes/commit/d30db1f9a915aa95402e1190461469a1889d92be) is one of the most illustrative. Another good reference is how Apache httpd uses its license: [The Apache license text remains intact](http://svn.apache.org/viewvc/httpd/httpd/trunk/LICENSE?revision=1824973&view=markup#l189) but a copyright notice should be displayed in every non-trivial source file. This is also consistent with [Apache license's usage guide](http://www.apache.org/dev/apply-license.html#new), which states nothing about changing `LICENSE`:

> 
> To apply the ALv2 to a new software distribution, include one copy of the license text by copying the file:
> 
> http://www.apache.org/licenses/LICENSE-2.0.txt
> 
> into a file called LICENSE in the top directory of your distribution. If the distribution is a jar or tar file, try to add the LICENSE file first in order to place it at the top of the archive. This covers the collective licensing for the distribution.
> 
> In addition, a correct NOTICE file MUST be included in the same directory as the LICENSE file.
> 
> Each original source document (code and documentation, but excluding the LICENSE and NOTICE files) SHOULD include a short license header at the top. If the distribution contains documents not covered by CLA, CCLA or Software Grant (such as third-party libraries) then see the policy guide.

But, on the other hand, I think adding a copyright notice and a short license header to every non-trivial file is necessary.

**Comment by splovyt** [comment]
So i have added the headers here, but will undo the changes to the LICENSE file as well.

**Comment by ajbozarth** [comment]
Ah yes, sorry, went into tunnel vision with my comments. I meant to point out that the headers should use that line for the copyright. Based on your sources I wholly agree with leaving the root LICENSE file unchanged.

File: .dockerignore [hunk 2]
```diff
@@ -1,2 +1,18 @@
+#
```
Comment (ajbozarth) [inline .dockerignore]: no need for license in ignore files

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,19 @@
+#
```
Comment (ajbozarth) [inline .gitignore]: no need for license in ignore files

**Comment by splovyt** [comment]
Good to have everyone on the same page now 🙂 thanks everyone

**Comment by xuhdev** [comment]
@ajbozarth No problem -- Just to clarify

**Comment by splovyt** [comment]
> Looks like the year should be `2019` or `2018-2019`?

I personally don't know the rules around changing the copyright year, but I made the change..

**Comment by ajbozarth** [comment]
I'd go with `2018-2019` given most of our models were released in 2018 but were adding the headers in 2019

**Comment by splovyt** [comment]
Sure, I have amended the commit 🙂

**Comment by ptitzler** [comment]
The year the code was first released. You can, but don't have to add the current year.

**Comment by splovyt** [comment]
Should we move forward with this PR and merge?

**Pull request merged**